### PR TITLE
Fix local proto imports

### PR DESCRIPTION
**Pull request by spenczar**
*Issue #, if available:*
#196 

*Description of changes:*
These are some subtle changes to how we map `.proto` files to package names. It should resolve the bug in #196 by correctly identifying when an imported `.proto` file has the same Go import path as the file we're generating.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by spenczar** [comment]
This is only a partial fix for #196 at present; there are multiple issues there.

**Comment by spenczar** [comment]
This should now be a complete fix.

File: internal/twirptest/importer_local/importer_local.proto [hunk 1]
```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+
+package twirp.internal.twirptest.importer_local;
+option go_package = "github.com/twitchtv/twirp/internal/twirptest/importer_local";
+
+import "importer_local_msgdef.proto";
+
+service Svc {
```
Comment (dpolansky) [inline internal/twirptest/importer_local/importer_local.proto]: Could you add a comment (wherever you deem appropriate) on what this package is testing for future reference?

File: protoc-gen-twirp/generator.go [hunk 2]
```diff
@@ -114,22 +114,42 @@ func (t *twirp) Generate(in *plugin.CodeGeneratorRequest) *plugin.CodeGeneratorR
 	}
 	t.genPkgName = genPkgName
 
+	// We also need to figure out the fully import path of the package we're
```
Comment (dpolansky) [inline protoc-gen-twirp/generator.go]: The comments help tremendously with understanding the nuance of how package names are assigned -- thanks!

**Pull request merged**